fix ARM64 test panics

meowsbits opened the pull request:
This fixes #346, and includes some extras.

- The fix: 69c9cc0
- Other are extras, but related and good to have overall I think. (Except dd4fea0, the rest are commits that we'll merge in eventually anyway).

The cherry-picked `go.mod` changes required conflict resolution.

holiman committed:
cmd/geth: make tests run quicker + use less memory and disk (#21919)

Signed-off-by: meows <b5c6@protonmail.com>

holiman committed:
go.mod: update github.com/golang/snappy(#21934)

This updates the snappy library depency to include a fix for
a Go 1.16 incompatibility issue.

Signed-off-by: meows <b5c6@protonmail.com>

alex347 committed:
cmd/geth: fix js unclean shutdown (#22302)

Signed-off-by: meows <b5c6@protonmail.com>

ucwong committed:
go.mod: upgrade goleveldb to commit 64b5b1c (#22436)

This pulls in a fix for a corruption issue when the process crashes
while a new manifest file is being added.

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/geth: move CoreGeth-specific test to non-conflicting file

Date: 2021-03-19 10:44:55-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
go.sum: run 'go get -t ./...'

Date: 2021-03-20 11:35:43-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr commented on cmd/geth/accountcmd_test.go:
```diff
@@ -43,13 +43,13 @@ func tmpDatadirWithKeystore(t *testing.T) string {
 
 func TestAccountListEmpty(t *testing.T) {
-	geth := runGeth(t, "account", "list")
+	geth := runGeth(t, "--nousb", "account", "list")
```
I might remove the `--nousb` if not really needed, as in the next go-ethereum merge, the revert the flag to `--usb`

meowsbits replied:
Done with https://github.com/etclabscore/core-geth/pull/347/commits/c5fe6a635aab6df080197113dd078dfd7eec9603

ziogaschr commented on cmd/geth/consolecmd.go:
```diff
@@ -231,13 +229,10 @@ func ephemeralConsole(ctx *cli.Context) error {
 
 	go func() {
-		<-abort
-		os.Exit(0)
```
Was the issue indeed the `os.Exit(0)` on latest go?

meowsbits replied:
No... it was the `snappy` library. https://github.com/etclabscore/core-geth/commit/69c9cc008d4b742dddb31f138b201547e91812ab

meowsbits committed:
cmd/geth: remove --nousb flag from runGeth tests

Resolves https://github.com/etclabscore/core-geth/pull/347#discussion_r598162353.

Date: 2021-03-21 07:37:35-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge branch 'master' into core-geth-issue-346-fix

Conflicts:
      cmd/geth/consolecmd_test.go

meowsbits committed:
cmd/geth: resolve errors caused by merge with 119fe1dd

Moves core-geth-specific tests to non-conflicting
file.

Date: 2021-03-21 07:46:25-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr reviewed (approved):
LGTM! I don't have an ARM64 device to give it a test, so I trust you and @iquidus for testing it.
